mlx/c/array.h

```diff
@@ -247,6 +247,12 @@ const float16_t* mlx_array_data_float16(mlx_array arr);
  */
 const bfloat16_t* mlx_array_data_bfloat16(mlx_array arr);
 #endif
+
+/**
+ * Replaces the array descriptor with that of another array.
+ */
```

Comment:
I'm ok exposing this as it is indeed necessary for "in place" bindings. But it might be good to add a little warning to discourage general use. Something like:

```
Note, this function is only intended for internal use in MLX bindings built with MLX C.
```

Wdyt?

Reply:
I ended up not using this directly because I can directly change pointer location to that of the result of `add()`. The note looks good though! What do you suggest? Do we close this until another language binding needs it (we're doing Rust). Or do we add it anyway with this note?

Reply:
Ah, in that case let's close this. We can always open it if we need this binding in the future.